### PR TITLE
README editing pass

### PR DESCRIPTION
**Pull request by mhilbrunner**
A few editing improvements/fixes to the README text.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,18 +80,18 @@ After following either one of these steps the site will be available at `http://
...
+Note that this is not relevant for local development. Locally you would build the website in place and then serve the `_site`
 folder. See the detailed instructions above.
 
 ## Project structure
 
 ### Content data and metadata
 
 The following folders contain data files, used for generating more dynamic parts of the website, such as the blog,
-the showcase, the downloads page. These pages are written in Markdown and contain a metadata header used by the
+the showcase and the downloads page. These pages are written in Markdown and contain a metadata header used by the
```
Comment (AThousandShips) [inline README.md]:
```suggestion
the showcase, and the downloads page. These pages are written in Markdown and contain a metadata header used by the
```
Oxford comma

File: README.md [hunk 2]
```diff
@@ -206,13 +206,13 @@ The `stable` key refers to hosts available for the stable release of that versio
...
+For the logic side of things, the new host needs to be supported by the `_plugins/make_download.rb` script. Refer
 to how other hosts are handled in that file and do the necessary adjustments. We assume that the final filenames
 are standard across all hosts, so `_data/download_configs.yml` is respected.
 
 ## Resources
 
 - Join the discussion on Godot Contributors Chat in the [#website](https://chat.godotengine.org/channel/website)
   channel.
-- Please, consider the [website usage stats](https://stats.tuxfamily.org/godotengine.org) when relying on modern
+- Please consider [website usage stats](https://stats.tuxfamily.org/godotengine.org) when relying on modern
```
Comment (akien-mga) [inline README.md]: Since we're no longer hosting the website on TuxFamily, I'm not sure how relevant this still is.

**Review by akien-mga — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Calinou) [reply README.md]: I think that line can be removed entirely, or we can point to https://caniuse.com/.

Comment (YuriSizov) [reply README.md]: Seconding caniuse as a de facto standard.

File: README.md [hunk 3]
```diff
@@ -80,18 +80,18 @@ After following either one of these steps the site will be available at `http://
 ### Deployment
 
 The project is built automatically by GitHub Actions whenever the `master` branch receives a new commit. The `master` branch
-itself should not be deployed, as it only contains the source files. The build version of the website is available as the
+itself should not be deployed, as it only contains the source files. The built version of the website is available on the
```
Comment (YuriSizov) [inline README.md]: I think "as" is okay, btw, as the branch is the direct representation of the built version 🙃

**Comment by mhilbrunner** [comment]
Adressed all feedback. Thanks y'all :)

File: README.md (hunk 2 above)
Comment (mhilbrunner) [reply README.md]: Yeah that makes sense, I missed the URL it links to - changed to caniuse.com.

**Comment by coppolaemilio** [comment]
Thanks!

**Pull request merged**